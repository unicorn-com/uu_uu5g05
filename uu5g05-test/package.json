{
  "name": "uu5g05-test",
  "version": "1.40.1",
  "description": "The library contains tools for testing of a components.",
  "license": "GPL-3.0",
  "namespace": "Uu5Test",
  "spec": "nodejs-lib; uu_appg01_devkit@6.6.0; uu5devkitg01-plugin@1.5.1",
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.6.3",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.6.1",
    "fake-indexeddb": "6.0.0",
    "jest-localstorage-mock": "2.4.26",
    "node-fetch": "2.6.7",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "react-test-renderer": "18.3.1"
  },
  "devDependencies": {
    "eslint-plugin-jest-dom": "5.5.0",
    "eslint-plugin-testing-library": "6.5.0",
    "nyc": "15.1.0",
    "uu_appg01_devkit": "^6.6.0",
    "uu5devkitg01-plugin": "^1.5.1"
  },
  "engines": {
    "node": ">=12.0.0"
  },
  "scripts": {
    "clean": "uu_appg01_devkit clean",
    "test": "uu_appg01_devkit testBrowser",
    "build": "uu_appg01_devkit build",
    "package": "uu_appg01_devkit package",
    "start": "uu_appg01_devkit start",
    "prettify": "uu_appg01_devkit prettify",
    "vulnerabilityScan": "uu_appg01_devkit vulnerabilityScan",
    "uuAppDevKitUpgrade": "uu_appg01_devkit uuAppDevKitUpgrade",
    "uu5Generate": "uu5devkitg01-plugin generate"
  },
  "main": "src/index.js",
  "files": [
    "src",
    "dist*",
    "test/setup"
  ],
  "prettier": {
    "printWidth": 120
  }
}
