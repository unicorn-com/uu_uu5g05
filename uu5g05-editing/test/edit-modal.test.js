import Uu5Editing from "uu5g05-editing";
// import Uu5Elements from "uu5g05-elements";
import { testProperties, shallow } from "uu5g05-test";

const CONFIG = {
  props: {
    props: {
      colorScheme: "primary",
      significance: "highlighted",
      nestingLevel: "area",
      card: "full",
      borderRadius: "moderate",
    },
    defaultProps: {
      card: "none",
    },
  },
  requiredProps: {
    open: true,
    uu5Tag: "Block",
    props: {
      colorScheme: "primary",
      significance: "highlighted",
      nestingLevel: "area",
      card: "full",
      borderRadius: "moderate",
    },
    onClose: jest.fn((e) => e?.persist?.()),
    onSave: jest.fn((e) => e?.persist?.()),
  },
  opt: {},
};

describe(`Uu5Editing.EditModal`, () => {
  testProperties(Uu5Editing.EditModal, CONFIG);

  it("onClose - should call onClose function", () => {
    let onClose = jest.fn((e) => e?.persist?.());
    let onSave = jest.fn((e) => e?.persist?.());

    let wrapper = shallow(
      <Uu5Editing.EditModal
        open={true}
        uu5Tag={"Block"}
        onClose={onClose}
        onSave={onSave}
        props={{
          colorScheme: "primary",
          significance: "highlighted",
          nestingLevel: "area",
          card: "full",
          borderRadius: "moderate",
        }}
        tabList={["visual"]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
