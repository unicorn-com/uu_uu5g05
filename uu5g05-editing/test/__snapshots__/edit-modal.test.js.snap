// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uu5Editing.EditModal default props 1`] = `
<Uu5Forms.withValidationMap(Uu5Forms.FormProvider)
  initialValue={
    Object {
      "borderRadius": "moderate",
      "card": "full",
      "colorScheme": "primary",
      "nestingLevel": "area",
      "significance": "highlighted",
    }
  }
  key="1"
  onSubmit={[Function]}
  preserveValueOnUnmount={true}
>
  <Component />
</Uu5Forms.withValidationMap(Uu5Forms.FormProvider)>
`;

exports[`Uu5Editing.EditModal defaultProps=undefined 1`] = `
<Uu5Forms.withValidationMap(Uu5Forms.FormProvider)
  initialValue={
    Object {
      "borderRadius": "moderate",
      "card": "full",
      "colorScheme": "primary",
      "nestingLevel": "area",
      "significance": "highlighted",
    }
  }
  key="1"
  onSubmit={[Function]}
  preserveValueOnUnmount={true}
>
  <Component />
</Uu5Forms.withValidationMap(Uu5Forms.FormProvider)>
`;

exports[`Uu5Editing.EditModal onClose - should call onClose function 1`] = `
<Uu5Forms.withValidationMap(Uu5Forms.FormProvider)
  initialValue={
    Object {
      "borderRadius": "moderate",
      "card": "full",
      "colorScheme": "primary",
      "nestingLevel": "area",
      "significance": "highlighted",
    }
  }
  key="1"
  onSubmit={[Function]}
  preserveValueOnUnmount={true}
>
  <Component />
</Uu5Forms.withValidationMap(Uu5Forms.FormProvider)>
`;

exports[`Uu5Editing.EditModal props=undefined 1`] = `
<Uu5Forms.withValidationMap(Uu5Forms.FormProvider)
  initialValue={Object {}}
  key="1"
  onSubmit={[Function]}
  preserveValueOnUnmount={true}
>
  <Component />
</Uu5Forms.withValidationMap(Uu5Forms.FormProvider)>
`;
